### PR TITLE
Format typespec on save

### PR DESCRIPTION
**Pull request by timotheeguerin**
Try to reduce people having formatting issue in typespec prs

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -27,5 +27,9 @@
   "[typescript]": {
     "editor.defaultFormatter": "esbenp.prettier-vscode",
     "editor.formatOnSave": true
+  },
+  "[typespec]": {
+    "editor.defaultFormatter": "typespec.typespec-vscode",
```
Comment (mikeharder) [inline .vscode/settings.json:32]: Is this formatter pre-installed by the `package-lock.json` in repo root, or is this something devs will need to install manually?

Comment (timotheeguerin) [reply .vscode/settings.json:32]: this is the typespec vscode extension that provide the format action.
The extension then use the language server provided in the locally installed compiler

Comment (mikeharder) [reply .vscode/settings.json:32]: Will it work like this?

1. Devs can optionally choose to install the typespec vscode extension
2. If installed, this enables formatOnSave
3. In not installed, this is a no-op

Comment (timotheeguerin) [reply .vscode/settings.json:32]: on save it is a noop if the extension is not here, if you explicitly do format then it shows this popup
image

Comment (timotheeguerin) [reply .vscode/settings.json:32]: should actually add the typespec extension as a recommendation i'll do that

**Review by mikeharder — APPROVED** [review]

**Pull request merged**